workitem/link/link_repository_blackbox_test.go

```diff
@@ -187,6 +190,216 @@ func (s *linkRepoBlackBoxTest) TestCreate() {
+		//
+		// Scenarios
+		// ---------
+		//
+		//   I:        II:       III:      IV:       V:
+		//
+		//    C         C         C         C         A
+		//     *         \         *         *         \
+		//      A         A         A         A         B
+		//       \         \         \         \         *
+		//        B         B         C         B         C
+		//         \         *         \
+		//          C         C         B
+		//
+		// I, II, III are cycles
+		// IV and V are no cycles.
+
+		// Map topologies to expected error during cycle
+		topos := map[link.Topology][5]bool{
+			link.TopologyNetwork:         {false, false, false, false, false},
+			link.TopologyDirectedNetwork: {false, false, false, false, false},
+			link.TopologyTree:            {true, true, true, false, false},
+			link.TopologyDependency:      {true, true, true, false, false},
+		}
+		for topo, errorExpected := range topos {
+			t.Run("topology: "+topo.String(), func(t *testing.T) {
+				//   I:
+				//
+				//    C
+				//     *
+				//      A
+				//       \
+				//        B
+				//         \
+				//          C
+				t.Run("Scenario I", func(t *testing.T) {
+					// given
+					fxt := tf.NewTestFixture(t, s.DB,
+						tf.WorkItems(3, tf.SetWorkItemTitles("A", "B", "C")),
+						tf.WorkItemLinkTypes(1, tf.SetTopologies(topo)),
+						tf.WorkItemLinks(2, func(fxt *tf.TestFixture, idx int) error {
+							l := fxt.WorkItemLinks[idx]
+							switch idx {
+							case 0:
+								l.SourceID = fxt.WorkItemByTitle("A").ID
+								l.TargetID = fxt.WorkItemByTitle("B").ID
+							case 1:
+								l.SourceID = fxt.WorkItemByTitle("B").ID
+								l.TargetID = fxt.WorkItemByTitle("C").ID
+							}
+							return nil
+						}),
+					)
+					// when
+					_, err := s.workitemLinkRepo.Create(s.Ctx, fxt.WorkItemByTitle("C").ID, fxt.WorkItemByTitle("A").ID, fxt.WorkItemLinkTypes[0].ID, fxt.Identities[0].ID)
+					// then
+					if errorExpected[0] {
+						require.Error(t, err)
+					} else {
+						require.NoError(t, err)
+					}
+				})
+				//   II:
+				//
+				//    C
+				//     \
+				//      A
+				//       \
+				//        B
+				//         *
+				//          C
+				t.Run("Scenario II", func(t *testing.T) {
+					// given
+					fxt := tf.NewTestFixture(t, s.DB,
+						tf.WorkItems(3, tf.SetWorkItemTitles("C", "A", "B")),
+						tf.WorkItemLinkTypes(1, tf.SetTopologies(topo)),
+						tf.WorkItemLinks(2, func(fxt *tf.TestFixture, idx int) error {
+							l := fxt.WorkItemLinks[idx]
+							switch idx {
+							case 0:
+								l.SourceID = fxt.WorkItemByTitle("C").ID
+								l.TargetID = fxt.WorkItemByTitle("A").ID
+							case 1:
+								l.SourceID = fxt.WorkItemByTitle("A").ID
+								l.TargetID = fxt.WorkItemByTitle("B").ID
+							}
+							return nil
+						}),
+					)
+					// when
+					_, err := s.workitemLinkRepo.Create(s.Ctx, fxt.WorkItemByTitle("B").ID, fxt.WorkItemByTitle("C").ID, fxt.WorkItemLinkTypes[0].ID, fxt.Identities[0].ID)
```

Comment:
@kwk Shouldn't the sourceID here be workitem C? 
In scenario 1, we are trying to link C to A -> https://github.com/fabric8-services/fabric8-wit/pull/1810/files#diff-bf110de3daad9b25035f4012579bbe55R258
In scenario 2, we are trying to link C to B (as depicted in the above diagram)

Reply:
@DhritiShikhar in the diagram above the link goes from B to C, so my code is correct. C shall be the child of B.